src/js/shared_controls.js

```diff
@@ -692,6 +697,9 @@ function calcStat(poke, statName) {
 		if (statName !== "hp") nature = poke.find(".nature").val();
 	}
 	var total = calc.calcStat(gen, legacyStatToStat(statName), base, ivs, evs, level, nature);
+	if (gen >= 8 && statName === "hp" && poke.isMax) {
```

Comment:
I would check `gen === 8` since Gamefreak seems to be adding new elements into one generation and abandoning it after that

Reply:
Honestly, if you're checking `.isMax`, do you also need the gen check?

Reply:
Good point, `isMax` is good enough